planner/core/testdata/integration_suite_out.json

```diff
@@ -881,5 +881,88 @@
+          "├─TableReader_25(Build) 3.00 root  data:TableFullScan_24",
+          "│ └─TableFullScan_24 3.00 cop[tikv] table:t1 keep order:false",
+          "└─TableReader_9(Probe) 1.00 root  data:TableRangeScan_8",
+          "  └─TableRangeScan_8 1.00 cop[tikv] table:t2 range: decided by [test.t.a], keep order:true"
+        ],
+        "Res": [
+          "1 111 1.1000000000 11 1 111 1.1000000000 11",
+          "2 222 2.2000000000 12 2 222 2.2000000000 12",
+          "3 333 3.3000000000 13 3 333 3.3000000000 13"
+        ]
+      },
+      {
+        "SQL": "select /*+ inl_hash_join(t1, t2) */ * from t t1 join t t2 on t1.a = t2.a",
+        "Plan": [
+          "IndexHashJoin_14 3.00 root  inner join, inner:TableReader_7, outer key:test.t.a, inner key:test.t.a",
+          "├─TableReader_25(Build) 3.00 root  data:TableFullScan_24",
+          "│ └─TableFullScan_24 3.00 cop[tikv] table:t1 keep order:false",
+          "└─TableReader_7(Probe) 1.00 root  data:TableRangeScan_6",
+          "  └─TableRangeScan_6 1.00 cop[tikv] table:t2 range: decided by [test.t.a], keep order:false"
+        ],
+        "Res": [
+          "1 111 1.1000000000 11 1 111 1.1000000000 11",
+          "2 222 2.2000000000 12 2 222 2.2000000000 12",
+          "3 333 3.3000000000 13 3 333 3.3000000000 13"
+        ]
+      },
+      {
+        "SQL": "select /*+ inl_join(t1, t2) */ * from t t1 join t t2 on t1.a = t2.a and t1.b = t2.b",
+        "Plan": [
+          "IndexJoin_10 3.00 root  inner join, inner:TableReader_7, outer key:test.t.a, test.t.b, inner key:test.t.a, test.t.b",
+          "├─TableReader_25(Build) 3.00 root  data:TableFullScan_24",
+          "│ └─TableFullScan_24 3.00 cop[tikv] table:t1 keep order:false",
+          "└─TableReader_7(Probe) 1.00 root  data:TableRangeScan_6",
+          "  └─TableRangeScan_6 1.00 cop[tikv] table:t2 range: decided by [test.t.a test.t.b], keep order:false"
+        ],
+        "Res": [
+          "1 111 1.1000000000 11 1 111 1.1000000000 11",
+          "2 222 2.2000000000 12 2 222 2.2000000000 12",
+          "3 333 3.3000000000 13 3 333 3.3000000000 13"
+        ]
+      },
+      {
+        "SQL": "select /*+ inl_join(t1, t2) */ * from t t1 join t t2 on t1.c = t2.c",
+        "Plan": [
+          "IndexJoin_11 3.00 root  inner join, inner:IndexLookUp_10, outer key:test.t.c, inner key:test.t.c",
+          "├─TableReader_35(Build) 3.00 root  data:Selection_34",
+          "│ └─Selection_34 3.00 cop[tikv]  not(isnull(test.t.c))",
+          "│   └─TableFullScan_33 3.00 cop[tikv] table:t1 keep order:false",
+          "└─IndexLookUp_10(Probe) 1.00 root  ",
+          "  ├─Selection_9(Build) 1.00 cop[tikv]  not(isnull(test.t.c))",
+          "  │ └─IndexRangeScan_7 1.00 cop[tikv] table:t2, index:c(c) range: decided by [eq(test.t.c, test.t.c)], keep order:false",
+          "  └─TableRowIDScan_8(Probe) 1.00 cop[tikv] table:t2 keep order:false"
```

Comment:
May we need to change the name of `TableRowIDScan`, it's confusing when the primary key is not the int handle.

Reply:
Yes, it's a TODO for the clustered index.